src/SIOFrameReader.cc

```diff
@@ -66,6 +66,12 @@ std::unique_ptr<SIOFrameData> SIOFrameReader::readNextEntry(const std::string& n
                                         tableInfo._uncompressed_length);
 }
 
+std::unique_ptr<SIOFrameData> SIOFrameReader::readEntry(const std::string& name, const unsingned entry);
+  m_nameCtr[name] = entry;
```

Comment:
This is a very clever trick. Can you add a short comment that says that this will indeed work (in all cases) and that all the necessary checks are actually done in `readNextEntry`, because otherwise we will probably be wondering about this everytime we come across it.

Reply:
Added a comment